nix/default.nix

```diff
@@ -365,7 +365,7 @@ let
 
         minimumOCamlVersion = "4.07";
 
-        patches = [ branchInfo.patchFile ];
+        patches = [ branchInfo.patches ];
```

Comment:
AFAIU it was a list of patches (with one item), but now it is a list of lists of patches, but I barely know `nix`. Can you clarify where it is equivalent? If it was Haskell I would say it should be `patches = branchInfo.patches`

Reply:
:think: this looks like a bug, but nix somehow figured it out

Reply:
So I'd like to know whether nix accidentally figured it out or it is reliable and specified behavior.

Reply:
I assume, that it performs list flattening somewhere, but I haven't found any proofs yet

Reply:
Anyway, don't you think that `patches = branchInfo.patches` would be less confusing?

Reply:
yeah, I agree